File: src/main/java/com/machfour/macros/names/ColumnNamer.java
package com.machfour.macros.names;

/*
 * Interface to provide user-readable names (strings) for nutrition data columns
 */

import com.machfour.macros.core.Column;
import com.machfour.macros.objects.NutritionData;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;

public interface ColumnNamer {
    @NotNull
    String getNutrientName(Column<NutritionData, Double> col);

    @NotNull
    String getName(Column<?, ?> col);
}

File: src/main/kotlin/com/machfour/macros/names/ColumnNamer.kt
package com.machfour.macros.names

import com.machfour.macros.core.Column

/*
 * Interface to provide user-readable names (strings) for nutrition data columns
 */
interface ColumnNamer {
    fun getName(col: Column<*, *>): String

    // JVM default
    fun getAbbreviatedName(col: Column<*, *>): String {
        return getName(col)
    }
}